adapters/repos/db/lsmkv/segment_serialization_benchmarks_test.go

```diff
@@ -63,3 +71,310 @@ func BenchmarkCollectionNodeKeyIndexAndWriteTo(b *testing.B) {
+			b.Fatal("error reading file:", err)
+		}
+
+		if len(all) == 0 {
+			b.Fatalf("file is empty")
+		}
+
+		err = ParseReplaceNodeIntoPread(bytes.NewReader(all), uint16(tc.secondaryKeysCount), out)
+		if err != nil {
+			b.Fatal("error parsing test data:", err)
+		}
+	}
+}
+
+// runDirectFileAccess measures performance of direct file access using multiple `pread` calls.
+// Includes the overhead of system calls for reading from disk.
+func runDirectFileAccess(b *testing.B, tc struct {
+	name               string
+	valueSize          int
+	keySize            int
+	secondaryKeysCount int
+	secondaryKeySize   int
+}, data []byte, tempFile string, out *segmentReplaceNode) {
+	file, cleanup := openFile(b, tempFile)
+	defer cleanup()
+
+	b.ResetTimer()
+	b.SetBytes(int64(len(data)))
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		if _, err := file.Seek(0, 0); err != nil {
+			b.Fatal("Failed to seek file:", err)
+		}
+
+		err := ParseReplaceNodeIntoPread(file, uint16(tc.secondaryKeysCount), out)
+		if err != nil {
+			b.Fatal("error parsing test data:", err)
+		}
+	}
+}
+
+func generateTestData(valueSize, keySize, secondaryKeysCount, secondaryKeySize int) ([]byte, error) {
+	buffer := new(bytes.Buffer)
+	var err error
+
+	if err = binary.Write(buffer, binary.LittleEndian, byte(rand.Intn(2))); err != nil {
+		return nil, fmt.Errorf("error writing tombstone binary: %w", err)
+	}
+
+	if err = binary.Write(buffer, binary.LittleEndian, uint64(valueSize)); err != nil {
+		return nil, fmt.Errorf("error writing valueSize binary: %w", err)
+	}
+
+	valueBuffer := make([]byte, valueSize)
+	if _, err = crand.Read(valueBuffer); err != nil {
+		return nil, fmt.Errorf("error reading valueSize binary: %w", err)
+	}
+	buffer.Write(valueBuffer)
+
+	if err = binary.Write(buffer, binary.LittleEndian, uint32(keySize)); err != nil {
+		return nil, fmt.Errorf("error writing keySize binary: %w", err)
+	}
+
+	keyBuffer := make([]byte, keySize)
+	if _, err = crand.Read(keyBuffer); err != nil {
+		return nil, fmt.Errorf("error reading keySize binary: %w", err)
+	}
+	buffer.Write(keyBuffer)
+
+	for i := 0; i < secondaryKeysCount; i++ {
+		if err = binary.Write(buffer, binary.LittleEndian, uint32(secondaryKeySize)); err != nil {
+			return nil, fmt.Errorf("error writing secondaryKeySize binary: %w", err)
+		}
+
+		if secondaryKeySize > 0 {
+			secondaryKeyBuffer := make([]byte, secondaryKeySize)
+			if _, err := crand.Read(secondaryKeyBuffer); err != nil {
+				return nil, fmt.Errorf("error reading secondaryKeySize binary: %w", err)
+			}
+			buffer.Write(secondaryKeyBuffer)
+		}
+	}
+
+	return buffer.Bytes(), nil
+}
+
+func makeTempDir(b *testing.B) (string, func()) {
+	tempDir, err := os.MkdirTemp("", "benchmark-*")
+	if err != nil {
+		b.Fatal("Failed to create temp directory:", err)
+	}
+	cleanup := func() {
+		err := os.RemoveAll(tempDir)
+		if err != nil {
+			b.Fatalf("Failed to remove temp directory %q: %v", tempDir, err)
+		}
+	}
+	return tempDir, cleanup
+}
```

Comment:
suggestion: a good way to register cleanup routines is via [`b.Cleanup`](https://cs.opensource.google/go/go/+/go1.24.1:src/testing/testing.go;l=1190), which is like `defer` but more explicit in this context

suggestion-2: better yet, since you're only using the temp dir for the test, you could use [this util method `b.TempDir()`](https://cs.opensource.google/go/go/+/go1.24.1:src/testing/testing.go;l=1224), which will automatically clean up the created directory after the test run finishes.